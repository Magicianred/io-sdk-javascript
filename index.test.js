const main = require("./index").main
const fs = require('fs')

test("main form", () => {
    //console.log(main)
    let form = main({})
    //console.log(form.body.form)
    expect(form).toMatchSnapshot()
})

test("choose file form", () => {
    //console.log(main)
    let form = main({ "sourcetype": "file" })
    //console.log(form.body.form)
    expect(form).toMatchSnapshot()
})

test("file form return data", () => {
    let file = fs.readFileSync("data/data.xlsx").toString('base64')
    let data = main({ "file": file })
    expect(data).toMatchSnapshot()
})

test("choose webapi form", () => {
    //console.log(main)
    let form = main({ "sourcetype": "webapi" })
    //console.log(form.body.form)
    expect(form).toMatchSnapshot()
})

test("webapi form return data", () => {
    let file = fs.readFileSync("data/data.xlsx").toString('base64')
    let data = main({ "file": file })
    expect(data).toMatchSnapshot()
})

