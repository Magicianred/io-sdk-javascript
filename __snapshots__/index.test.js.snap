// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`choose file form 1`] = `Promise {}`;

exports[`choose webapi form 1`] = `Promise {}`;

exports[`file form return data 1`] = `Promise {}`;

exports[`main form 1`] = `Promise {}`;

exports[`webapi form return data 1`] = `Promise {}`;
